code/obj/item/toilets.dm

```diff
@@ -112,25 +112,20 @@ TYPEINFO(/obj/item/storage/toilet)
 			reset_anchored(M)
 			M.buckled = null
 			src.add_fingerprint(user)
-	if((src.clogged < 1) || (src.contents.len < 7) || (user.loc != src.loc))
+	if((src.clogged < 1) || (!src.storage.is_full()) || (user.loc != src.loc))
 		user.visible_message("<span class='notice'>[user] flushes [src].</span>", "<span class='notice'>You flush [src].</span>")
 		playsound(src, 'sound/effects/toilet_flush.ogg', 50, 1)
 
 
 #ifdef UNDERWATER_MAP
-		var/turf/source = get_turf(src)
-		if (source)
-			var/turf/target = locate(source.x,source.y,5)
-			for (var/thing in contents)
-				var/atom/movable/A = thing
-				A.set_loc(target)
+		for (var/obj/item/I as anything in src.storage.get_contents())
+			src.storage.transfer_stored_item(I, get_turf(src))
```

Comment:
This is doing a different thing- old behavior was locating the turf 'under' this on Z5 (trench) and flushing the item to there, new behavior is just dumping the item on the toilet's tile

Reply:
Oops, fixed